File: src/main/java/net/atos/client/zgw/zrc/model/RolListParameters.java
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.zgw.zrc.model;

import java.net.URI;

import jakarta.ws.rs.QueryParam;

import net.atos.client.zgw.shared.model.AbstractListParameters;
import nl.info.client.zgw.zrc.model.generated.BetrokkeneTypeEnum;

/**
 *
 */
public class RolListParameters extends AbstractListParameters {

    /**
     * URL-referentie naar de ZAAK.
     */
    @QueryParam("zaak")
    private URI zaak;

    /**
     * Type van de betrokkene
     */
    @QueryParam("betrokkeneType")
    private String betrokkeneType;

    /**
     * URL-referentie naar een roltype binnen het ZAAKTYPE van de ZAAK.
     */
    @QueryParam("roltype")
    private URI roltype;

    public RolListParameters(final URI zaak) {
        this.zaak = zaak;
    }

    public RolListParameters(final URI zaak, final URI roltype) {
        this.zaak = zaak;
        this.roltype = roltype;
    }

    public RolListParameters(final URI zaak, final URI roltype, final BetrokkeneTypeEnum betrokkeneType) {
        this.zaak = zaak;
        this.betrokkeneType = betrokkeneType.toString();
        this.roltype = roltype;
    }

    public URI getZaak() {
        return zaak;
    }

    public void setZaak(final URI zaak) {
        this.zaak = zaak;
    }

    public String getBetrokkeneType() {
        return betrokkeneType;
    }

    public void setBetrokkeneType(final BetrokkeneTypeEnum betrokkeneType) {
        this.betrokkeneType = betrokkeneType.toString();
    }

    public URI getRoltype() {
        return roltype;
    }

    public void setRoltype(final URI roltype) {
        this.roltype = roltype;
    }
}

File: src/main/kotlin/nl/info/client/zgw/zrc/model/RolListParameters.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.zgw.zrc.model

import jakarta.ws.rs.QueryParam
import net.atos.client.zgw.shared.model.AbstractListParameters
import nl.info.client.zgw.zrc.model.generated.BetrokkeneTypeEnum
import java.net.URI

class RolListParameters : AbstractListParameters {
    /**
     * URL-referentie naar de ZAAK.
     */
    @QueryParam("zaak")
    var zaak: URI? = null

    /**
     * Type van de betrokkene
     */
    @QueryParam("betrokkeneType")
    var betrokkeneType: String? = null
        private set

    /**
     * URL-referentie naar een roltype binnen het ZAAKTYPE van de ZAAK.
     */
    @QueryParam("roltype")
    var roltype: URI? = null

    constructor(zaak: URI) : super() {
        this.zaak = zaak
    }

    constructor(zaak: URI, roltype: URI) : super() {
        this.zaak = zaak
        this.roltype = roltype
    }

    constructor(zaak: URI, roltype: URI, betrokkeneType: BetrokkeneTypeEnum) : super() {
        this.zaak = zaak
        this.betrokkeneType = betrokkeneType.toString()
        this.roltype = roltype
    }

    fun setBetrokkeneType(betrokkeneType: BetrokkeneTypeEnum) {
        this.betrokkeneType = betrokkeneType.toString()
    }
}
